### PR TITLE
Add ToText class

### PR DESCRIPTION
**Pull request by mbj**
### Acceptance

* [ ] Green CI

### CODE REVIEW

File: src/Data/Conversion.hs [hunk 1]
```diff
@@ -164,6 +164,12 @@ instance Conversion Text String where
 instance Conversion String Text where
   convert = Text.unpack
 
+class Conversion Text a => ToText a where
+  toText :: a -> Text
+
+instance Conversion Text a => ToText a where
+  toText = convert
```
Comment (epicallan) [inline src/Data/Conversion.hs]: Why not have this as a default instance such that users can be free to define more custom instances if desired.  I also think you can't have `any class` derivations without `default` instances could confirm later.

**Review by epicallan — COMMENTED** [review]
Left one comment.

File: src/Data/Conversion.hs (hunk 1 above)
Comment (mbj) [reply src/Data/Conversion.hs]: good call.

But also your comment lets me think about something else: 

Why not:

```
type ToText a = Conversion Text a
```

And keep the original `toText`.

We do never want the `convert` and `toText` to differ IMO. Hence making a subclass which makes this possible is a bad idea.

`toText / ToText` should just be an alias.

Comment (epicallan) [reply src/Data/Conversion.hs]: > We do never want the convert and toText to differ IMO.

I like this rationale. Let's go for an alias.

**Pull request merged**